### PR TITLE
NPC debug menu update

### PR DESCRIPTION
**Pull request by Coolthulhu**
Moved the entire NPC debug menu (accessible with debug->d) to debug_menu.cpp.

Changed targeting: instead of having to manually find the NPC and "aim" at it, all available targets (player and NPCs) are listed in `uimenu`, similar to how chatting works.

Added a mission edit menu. It isn't perfect, but allows removing broken missions and testing mission success and failure without having to actually go through the missions.

Diff looks heavy, but 300 lines of it are just cut+paste from `game.cpp`.

### CODE REVIEW

File: src/debug_menu.cpp [hunk 1]
```diff
@@ -44,3 +69,515 @@ void debug_menu::teleport_overmap()
...
+                    p.reset_stats();
+                }
+            }
+        }
+        break;
+        case D_ITEMS:
+            wishitem( &p );
+            break;
+        case D_DELETE_ITEMS:
+            if( !query_yn( "Delete all items from the target?" ) ) {
```
Comment (illi-kun) [inline src/debug_menu.cpp]: Missed _()

**Comment by illi-kun** [comment]
Why both ```debug_menu.cpp``` and ```wish.cpp``` exist? I'm not asking to change it now, just trying to figure out the difference in their purposes.

File: src/game.cpp [hunk 2]
```diff
@@ -4199,323 +4199,8 @@ void game::debug()
...
-                        classes.query();
-                        if( classes.ret < (int)ids.size() && classes.ret >= 0 ) {
-                            np->randomize( ids[ classes.ret ] );
-                        }
-                    }
-                }
-            }
-        }
+        case 14:
+        debug_menu::npc_edit_menu();
```
Comment (mugling) [inline src/game.cpp]: Minor: bad indendation

File: src/debug_menu.cpp [hunk 3]
```diff
@@ -44,3 +69,515 @@ void debug_menu::teleport_overmap()
     const tripoint new_pos( omt_to_om_copy( g->u.global_omt_location() ) );
     add_msg( _( "You teleport to overmap (%d,%d,%d)." ), new_pos.x, new_pos.y, new_pos.z );
 }
+
+void npc_edit_menu()
+{
+    std::vector< tripoint > locations;
+    uimenu charmenu;
+    charmenu.return_invalid = true;
+    // Hack: uimenu doesn't like negative indices in entries
```
Comment (mugling) [inline src/debug_menu.cpp]: You should open that as a separate issue

File: src/debug_menu.cpp [hunk 4]
```diff
@@ -44,3 +69,515 @@ void debug_menu::teleport_overmap()
...
+    int charnum = 0;
+    charmenu.addentry( charnum++, true, MENU_AUTOASSIGN, "%s", _( "You" ) );
+    locations.emplace_back( g->u.pos() );
+    for( auto *npc_p : g->active_npc ) {
```
Comment (mugling) [inline src/debug_menu.cpp]: `auto *` is odd-looking syntax

**Comment by mugling** [comment]
I don't like that UI menu hack thing but this is much nicer for testing and developer QOL fixes are important if we hope to make any long-term progress

**Pull request merged**